client/src/components/Schedule/ListView.tsx

```diff
@@ -1,17 +1,219 @@
+
+    setStartWeekDate(startWeekText);
+    setEndWeekDate(endWeekText);
+  }, [currentWeek]);
+
+  const handleClickBack = () => {
+    setCurrentWeek(currentWeek - 1);
+  };
+
+  const handleClickForward = () => {
+    setCurrentWeek(currentWeek + 1);
+  };
+
+  const getCurrentDayKey = useMemo(() => {
+    const day = moment().day();
+    const currentDay = day ? day - 1 : 6;
+    const date = moment().format('YYYYMMDD');
+    const todaysEvents = data.filter(({ dateTime }) => date === moment(dateTime).format('YYYYMMDD'));
+
+    if (todaysEvents.length === 0) {
+      return [];
+    }
+
+    return `${WEEK[currentDay]}0`;
+  }, [data]);
+
   return (
-    <div>
-      <h2>List</h2>
-      Data length = {data.length}. Timezone = {timeZone}
+    <div className={'List'}>
+      <Row justify="center" align="middle" gutter={[16, 16]}>
+        <Col>
+          <Tooltip title="Previous week">
+            <Button shape="circle" icon={<LeftOutlined />} size="large" onClick={handleClickBack} />
+          </Tooltip>
+        </Col>
+        <Col>
+          <Text strong>{`${startWeekDate} - ${endWeekDate}`}</Text>
+        </Col>
+        <Col>
+          <Tooltip title="Next week">
+            <Button shape="circle" icon={<RightOutlined />} size="large" onClick={handleClickForward} />
+          </Tooltip>
+        </Col>
+      </Row>
+      <Collapse defaultActiveKey={getCurrentDayKey}>{weekElements(data, currentWeek, timeZone)}</Collapse>
+      <style jsx>{listStyles}</style>
     </div>
   );
-}
+};
+
+const getStartAndEndWeekTime = (currentWeek: number, timeZone: string) => {
+  const dayOfSelectedWeek = moment()
+    .tz(timeZone)
+    .add(currentWeek * weekLength, 'days');
+  const startWeek = dayOfSelectedWeek.clone().startOf('isoWeek');
+  const endWeek = dayOfSelectedWeek.clone().endOf('isoWeek');
+
+  return [startWeek, endWeek];
+};
+
+const isCurrentWeek = (date: string, timeZone: string, currentWeek: number) => {
+  const startAndEnd = getStartAndEndWeekTime(currentWeek, timeZone);
+  const milliseconds = moment(date).tz(timeZone).valueOf();
+
+  return milliseconds > startAndEnd[0].valueOf() && milliseconds < startAndEnd[1].valueOf();
+};
+
+const panelClassName = (dayOfWeek: number, currentWeek: number) => {
+  const today = moment().day();
+
+  if (currentWeek < 0 || (currentWeek === 0 && dayOfWeek < today)) {
+    return { backgroundColor: previuosDaysColor };
+  }
+
+  if (currentWeek > 0 || (currentWeek === 0 && dayOfWeek > today)) {
+    return { backgroundColor: nextDaysColor };
+  }
+
+  if (dayOfWeek === today) {
+    return { backgroundColor: currentDayColor };
+  }
+
+  return {};
+};
+
+const mapToWeek = (events: CourseEvent[], timeZone: string) => {
+  const weekMap = new Array(weekLength).fill([]);
+
+  events.forEach((event: CourseEvent) => {
+    let eventDay = moment(event.dateTime).tz(timeZone).day();
+
+    if (eventDay === 0) {
+      eventDay = 6;
+    } else {
+      eventDay -= 1;
+    }
```

Comment:
The same question about magic numbers here)

Reply:
fix